### PR TITLE
OCPBUGS-32397: Add links to FlowCollector API Reference in the NetObserv Configuration guide

### PR DESCRIPTION
**Pull request by skrthomas**
Version(s):

4.12+
Issue:

https://issues.redhat.com/browse/OCPBUGS-32397
Link to docs preview:

https://78138--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/network_observability/configuring-operator.html
QE review:
- QE not required. This is just making more prominent links to existing API documentation that is already in the Network Observability docs. I guess, a QE ACK would be nice but it isn't anything technically requiring hard review. 

Additional information:

### CODE REVIEW

File: modules/network-observability-flowcollector-view.adoc [hunk 1]
```diff
@@ -50,14 +50,15 @@ spec:
       conversationHeartbeatInterval: 30s
   loki:                     <4>
     mode: LokiStack         <5>
+    readTimeout: 3m0s       <6>
```
Comment (jpinsonneau) [inline modules/network-observability-flowcollector-view.adoc:53]: I would recommend to keep the `30s` default in the example here.

Comment (jpinsonneau) [reply modules/network-observability-flowcollector-view.adoc:53]: + consider adding the extra ones:
```suggestion
    readTimeout: 30s       <6>
    writeTimeout: 10s
    writeBatchWait: 1s
    writeBatchSize: 102400
    advanced:
      writeMinBackoff: 1s
      writeMaxBackoff: 5s
      writeMaxRetries: 2
      staticLabels: 
        app: netobserv-flowcollector
```

https://github.com/netobserv/network-observability-operator/blob/d9eb3694e04237c40e32ec679cc5bdb7d894b9d7/apis/flowcollector/v1beta2/flowcollector_types.go#L741-L763

Comment (skrthomas) [reply modules/network-observability-flowcollector-view.adoc:53]: Ok I will keep it to 30s. Idk about adding callouts for all these, but I can see adding them for visibility. Do you think they should all have code callouts? I know generally we want to de-emphasize Loki in the docs, but I guess I am inclined to add the `readTimeout` because someone asked for it in a bug.

Comment (jpinsonneau) [reply modules/network-observability-flowcollector-view.adoc]: Indeed, that's why the only loki option displayed in the doc here was the `mode`. I'm not especially in favor of adding too much informations here honnestly. 
Keeping the first install example simple makes things easy for the user to understand. We could even get rid of some sections such as `processor.advanced`, `resources`, `logLevel` etc. 

I would prefer having dedicated pages per component for extra explanations or just rely on the Custom Resource Definition as before.

@jotak any thoughts ?

Comment (skrthomas) [reply modules/network-observability-flowcollector-view.adoc]: Okay, thanks for the discussion around this. I will not document these Loki options here and will instead link to the API reference. As Joel points out, its not visible at all in the Configuring section, and maybe just linking there is sufficient. There's many configs in the CRD that we don't bring directly to this sample, and these Loki configs just don't seem to be **_the_** **_thing_** to bring here.

**Review by jotak — APPROVED** [review]
lgtm!

File: observability/network_observability/configuring-operator.adoc [hunk 2]
```diff
@@ -6,28 +6,29 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
-You can update the Flow Collector API resource to configure the Network Observability Operator and its managed components. The  Flow Collector is explicitly created during installation. Since this resource operates cluster-wide, only a single `FlowCollector` is allowed, and it has to be named `cluster`.
+You can update the `FlowCollector` API resource to configure the Network Observability Operator and its managed components. The  `FlowCollector` is explicitly created during installation. Since this resource operates cluster-wide, only a single `FlowCollector` is allowed, and it has to be named `cluster`. For more information, see the xref:../../observability/network_observability/flowcollector-api.adoc#network-observability-flowcollector-api-specifications_network_observability[FlowCollector API reference].
```
Comment (skopacz1) [inline observability/network_observability/configuring-operator.adoc:9]: Out of scope to your work, so it's up to you if you want to edit it, but per the [ISG word usage](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#h) entry on "have to", it should only be used when "must" is too restrictive, which I don't believe is the case here:
```suggestion
You can update the `FlowCollector` API resource to configure the Network Observability Operator and its managed components. The  `FlowCollector` is explicitly created during installation. Since this resource operates cluster-wide, only a single `FlowCollector` is allowed, and it must be named `cluster`. For more information, see the xref:../../observability/network_observability/flowcollector-api.adoc#network-observability-flowcollector-api-specifications_network_observability[FlowCollector API reference].
```

Comment (skopacz1) [inline observability/network_observability/configuring-operator.adoc:9]: Nit - the section that this link leads to technically uses "specifications" instead of reference. If this is intended to point to the specs sections, I'd update the link text:
```suggestion
You can update the `FlowCollector` API resource to configure the Network Observability Operator and its managed components. The  `FlowCollector` is explicitly created during installation. Since this resource operates cluster-wide, only a single `FlowCollector` is allowed, and it has to be named `cluster`. For more information, see the xref:../../observability/network_observability/flowcollector-api.adoc#network-observability-flowcollector-api-specifications_network_observability[FlowCollector API specifications].
```

If the link is intended to point to the overall assembly I'd update the link text _and_ the anchor:
```suggestion
You can update the `FlowCollector` API resource to configure the Network Observability Operator and its managed components. The  `FlowCollector` is explicitly created during installation. Since this resource operates cluster-wide, only a single `FlowCollector` is allowed, and it has to be named `cluster`. For more information, see the xref:../../observability/network_observability/flowcollector-api.adoc#flowcollector-api[FlowCollector configuration parameters].
```

In either case the "FlowCollector API reference" phrasing is only used in the topic map name for this page/file, should that match the assembly title?

File: observability/network_observability/configuring-operator.adoc [hunk 3]
```diff
@@ -6,28 +6,29 @@ include::_attributes/common-attributes.adoc[]
...
 
 
 include::modules/network-observability-flowcollector-view.adoc[leveloffset=+1]
 
 [role="_additional-resources"]
 .Additional resources
-For more information about conversation tracking, see xref:../../observability/network_observability/observing-network-traffic.adoc#network-observability-working-with-conversations_nw-observe-network-traffic[Working with conversations].
+* xref:../../observability/network_observability/flowcollector-api.adoc#network-observability-flowcollector-api-specifications_network_observability[FlowCollector API reference]
```
Comment (skopacz1) [inline observability/network_observability/configuring-operator.adoc:16]: Same comment as above about the title of the section this link points to

File: observability/network_observability/configuring-operator.adoc [hunk 4]
```diff
@@ -6,28 +6,29 @@ include::_attributes/common-attributes.adoc[]
...
+* xref:../../observability/network_observability/observing-network-traffic.adoc#network-observability-working-with-conversations_nw-observe-network-traffic[Working with conversations]
```
Comment (skopacz1) [inline observability/network_observability/configuring-operator.adoc:17]: Same comment about link text (although I recognize this is out of scope to your work, up to you):
```suggestion
* xref:../../observability/network_observability/observing-network-traffic.adoc#network-observability-working-with-conversations_nw-observe-network-traffic[Working with conversation tracking]
```

File: observability/network_observability/configuring-operator.adoc [hunk 5]
```diff
@@ -6,28 +6,29 @@ include::_attributes/common-attributes.adoc[]
...
 
 include::modules/network-observability-flowcollector-kafka-config.adoc[leveloffset=+1]
 include::modules/network-observability-enriched-flows.adoc[leveloffset=+1]
 
 [role="_additional-resources"]
 .Additional resources
-For more information about specifying flow format, see xref:../../observability/network_observability/json-flows-format-reference.adoc#network-observability-flows-format_json_reference[Network flows format reference].
+* xref:../../observability/network_observability/json-flows-format-reference.adoc#network-observability-flows-format_json_reference[Network flows format reference].
```
Comment (skopacz1) [inline observability/network_observability/configuring-operator.adoc:24]: Same comment:
```suggestion
* xref:../../observability/network_observability/json-flows-format-reference.adoc#network-observability-flows-format_json_reference[Network Flows format reference].
```

**Review by skopacz1 — COMMENTED** [review]
Just a few nits, otherwise LGTM!

File: observability/network_observability/configuring-operator.adoc (hunk 2 above)
Comment (skrthomas) [reply observability/network_observability/configuring-operator.adoc:9]: This is a great nit to catch. You're right, there's a little mis-matching in the names with the topic-map and assembly. Out of these suggestions, I think I will change the assembly title to "FlowCollector API reference".

File: observability/network_observability/configuring-operator.adoc (hunk 3 above)
Comment (skrthomas) [reply observability/network_observability/configuring-operator.adoc:16]: In the above comment, I responded with what I'm going to do to address this.

File: observability/network_observability/configuring-operator.adoc (hunk 5 above)
Comment (skrthomas) [reply observability/network_observability/configuring-operator.adoc:24]: I admittedly don't check this reference too often, but now that you bring my attn to this, I see there's a capitalization mismatch here too 🤦‍♀️ . I will update this: 
image
